jupyter/jsl_installation_script_ubuntu.sh

```diff
@@ -88,6 +92,131 @@ install_ocr(){
+    )
+    if response.ok:
+        data = response.json()
+        if "errors" in data:
+            raise Exception("Invalid token.")
+        licenses = [s["node"] for s in response.json()["data"]["licenses"]["edges"]]
+        return licenses
+    else:
+        raise Exception("Something went wrong...")
+
+
+def ensure_correct_choice(licenses_count):
+    license_id = input()
+    if license_id.isnumeric():
+        index = int(license_id) - 1
+        if licenses_count > index:
+            return index
+        else:
+            print(f"Please select value between 1 and {licenses_count}")
+            return ensure_correct_choice(licenses_count)
+    else:
+        print(f"Please select value between 1 and {licenses_count}")
+        return ensure_correct_choice(licenses_count)
+
+
+def get_user_license_choice(licenses):
+    print("Please select the license to use.")
+    for idx, license in enumerate(licenses):
+        products = ",".join(s["name"] for s in license["products"])
+        if license["platform"] is None:
+            scope = "Airgap"
+        else:
+            scope = license["platform"]["name"]
+            type = license["platform"]["type"]
+            if scope == "Floating":
+                if type:
+                    scope = scope + "," + type.capitalize()
+
+        print(
+            "{}. Libraries: {}\n   License Type: {}\n   Expiration Date: {}\n   Scope: {}".format(
+                idx + 1, products, license["type"], license["endDate"], scope
+            )
+        )
+
+    choice = ensure_correct_choice(len(licenses))
+    return licenses[choice]
+
+
+def download_license(license, access_token):
+    print("Downloading license...")
+    response = requests.get(
+        "{}/attachments/{}".format(MYJSL_ORIGIN,license["id"]),
+        headers={"Authorization": f"Bearer {access_token}"},
+    )
+    if response.ok:
+        license_file = sys.argv[2]
+        os.makedirs(os.path.dirname(license_file), exist_ok=True)
+        open(license_file, "wb").write(response.content)
+        print("Licenses extracted successfully")
+    else:
+        raise Exception(f"Failed fetching license. {response.reason}")
+
+
+if __name__ == "__main__":
+    try:
+        # get access token
+        token = sys.argv[1]
+        if token:
+            # get available licenses
+            licenses = get_user_licenses(token)
+            # if more than once license, show options
+            license_to_use = None
+            if len(licenses) == 0:
+                raise Exception(
+                    f"It seems there are no compatible licenses available. Please request a license first using {MYJSL_ORIGIN}"
+                )
+            if len(licenses) == 1:
+                license_to_use = licenses[0]
+            else:
+                license_to_use = get_user_license_choice(licenses)
+            # download license to file
+            download_license(license_to_use, token)
```

Comment:
But we need a license file for when the user selects the run (-r) flag to run this file.
Should we fetch a license each time user wants to run this script? (also the token can be expired sooner than license and this makes inconvenience for user)

Reply:
Let's download env variables everytime when the `-t` flag is specified.
If we download env variables to a file, it would be not clear, how to refetch them from the server.